packages/fortinet_fortimail/data_stream/log/elasticsearch/ingest_pipeline/pipeline_mail.yml

```
@@ -0,0 +1,133 @@
+---
+description: Pipeline for processing Mail logs.
+processors:
+  - set:
+      field: event.category
+      value: [email]
+      override: true
+  - set:
+      field: event.outcome
+      value: success
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'webmail' && ctx.fortinet_fortimail.log.log_details?.status.toLowerCase() == 'success'
+  - set:
+      field: event.outcome
+      value: failure
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'webmail' && ctx.fortinet_fortimail.log.log_details?.status.toLowerCase() == 'failure'
+  - append:
+      field: event.category
+      value: 'authentication'
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'webmail'
+      allow_duplicates: false
+  - set:
+      field: event.type
+      value: [info]
+  - append:
+      field: event.type
+      value: 'end'
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'webmail'
+  - append:
+      field: fortinet_fortimail.log.user
+      value: '{{{fortinet_fortimail.log.log_details.user}}}'
+      allow_duplicates: false
+      if: ctx.fortinet_fortimail?.log?.log_details?.user != null
+  - append:
+      field: user.name
+      value: '{{{fortinet_fortimail.log.log_details.user}}}'
+      allow_duplicates: false
+      if: ctx.fortinet_fortimail?.log?.log_details?.user != null
+  - append:
+      field: related.user
+      value: '{{{fortinet_fortimail.log.log_details.user}}}'
+      allow_duplicates: false
+      if: ctx.fortinet_fortimail?.log?.log_details?.user != null
+  - grok:
+      field: message
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'webmail'
+      patterns:
+        - "^%{DATA}[U|u]ser %{NOTSPACE:_tmp.user} %{DATA}%{IP:fortinet_fortimail.log.ip}%{GREEDYDATA:_tmp.msg}$"
+        - "^%{DATA}[L|l]ogin for \\'%{NOTSPACE:_tmp.user}\\' %{DATA}%{IP:fortinet_fortimail.log.ip}%{GREEDYDATA:_tmp.msg}$"
+        - "^%{DATA}[U|u]ser %{NOTSPACE:_tmp.user} %{GREEDYDATA:_tmp.msg}$"
```

```suggestion
        - "^%{PREFIX} %{DATA}%{IP:fortinet_fortimail.log.ip}%{GREEDYDATA:_tmp.msg}$"
        - "^%{DATA}[Uu]ser %{NOTSPACE:_tmp.user} %{GREEDYDATA:_tmp.msg}$"
      pattern_definitions:
        PREFIX: "(?:%{DATA}[Uu]ser %{NOTSPACE:_tmp.user}|%{DATA}[Ll]ogin for \\'%{NOTSPACE:_tmp.user}\\')"
```
(untested)
Note also possibility of `(?i)`.